libs/ui/elements/src/lib/metadata-contact/metadata-contact.component.ts

```diff
@@ -34,6 +34,37 @@ export class MetadataContactComponent {
     )
   }
 
+  parseAddress(inputAddress) {
+    const addressParts = inputAddress.split(',').map((part) => part.trim())
+
+    const addressArray = []
+
+    for (let i = 0; i < addressParts.length; i++) {
+      const part = addressParts[i]
+      if (part.toLowerCase().includes('cs')) {
+        // Handle "CS Number" in a single line
+        addressArray.push(part)
+      } else if (part.match(/^\d{5}$/)) {
+        // Combine postcode and city in a single line
+        const postcodeCity = `${part} ${addressParts[i - 1]}`
+        // delete duplicate city
+        if (postcodeCity.includes(addressParts[i - 1])) {
+          addressArray.pop()
+        }
+        addressArray.push(postcodeCity)
+      } else {
+        // Treat as a separate line
+        addressArray.push(part)
+      }
```

Comment:
I'm not sure about this logic; it makes a lot of assumptions about the structure of an address, and honestly I don't think we can assume that much as address fields can often be filled completely freely by metadata authors; e.g. assuming that "CS" will be present before a postcode (which is specific to France I guess), assuming that the city name comes before the postal code...

Relying on the "," separator is enough IMO, and putting each part in a separate line.

Reply:
On the other hand it looks like the GN editor reliably produces this kind of output... but that might change anytime in the future